server/gc/src/main/java/org/apache/accumulo/gc/SimpleGarbageCollector.java

```diff
@@ -865,6 +897,12 @@ static boolean isDir(String delete) {
 
   @Override
   public GCStatus getStatus(TInfo info, TCredentials credentials) {
-    return status;
+    statusLock.lock();
+    try {
+      GCStatus copy = new GCStatus(status);
```

Comment:
I think these two lines are all you need to do for this change... make a copy and return. 

It seems like all you need is an immutable copy of the GCStatus object, which can be done without any locking.  The locking is not needed and just adds complexity.  No matter what you do, you will only ever be able to get a snapshot of the GC at a moment in time and we want to avoid making too many changes to 1.9.

You could just add a separate method like ```getSnapshot()``` which makes a copy of status.  This would make a merge forward easier as well.

Reply:
The code that I think that also needs to lock is when the "last run" values are set to the current values and the zero'ed for the next run.
`   status.lastLog = status.currentLog;
     status.currentLog = new GcCycleStats();`

The rest of the locking was to maintain current behavior - and to make it consistent and correct for places where the current code is not locking.  It seemed odd to have portions that are thread safe and other portions that require single-thread access for correct behavior. This seemed a better practice, especially when having access to the snapshot is done from other thread(s)